src/components/Icon/Icon.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { useState, useEffect, useRef } from 'react'
+import { icons } from '@/assets/icons'
+import {  IconType, IconProps } from '@/types/icon'
+
+function Icon(props: IconProps) {
+  const importedIconRef = useRef<IconType>()
+  const [loading, setLoading] = useState(false)
+  
+  const iconSizes = {
+    'sm': {
```

Comment:
Can we use the enum as the keys here please

Reply:
This should probably live outside the components function as a constant, e.g.

``` typescript
const ICON_SIZES = {
  [IconSize.Small]: {
    //...
  },
  //...
}

function Icon(props: IconProps) {
  //..
```

Reply:
👍